### PR TITLE
add snap integration tests for opensuse and remove snap in handlers

### PR DESCRIPTION
**Pull request by aminvakil**
##### SUMMARY
Extends https://github.com/ansible-collections/community.general/pull/2907
Also enables snap integration tests.

##### ISSUE TYPE
- Test Pull Request

##### COMPONENT NAME
plugins/modules/packaging/os/snap.py

### CODE REVIEW

File: tests/integration/targets/snap/tasks/main.yml [hunk 1]
```diff
@@ -4,20 +4,35 @@
 # and should not be used as examples of how to write Ansible roles #
 ####################################################################
 
-- name: install snapd
+- name: install snapd on ubuntu
   apt:
     name: snapd
     state: present
   register: snapd_install_ubuntu
```
Comment (felixfontein) [inline tests/integration/targets/snap/tasks/main.yml:11]: Can you use `notify:` and handlers to do the uninstall?

Comment (aminvakil) [reply tests/integration/targets/snap/tasks/main.yml:11]: Yes, I will change it in a couple of hours.

**Comment by felixfontein** [comment]
@aminvakil this now has conflicts. How do you want to proceed with this PR?

**Comment by aminvakil** [comment]
> @aminvakil this now has conflicts. How do you want to proceed with this PR?

It's clear that `snap` cannot be run on docker (at least not easily), so I guess if I want to add more OSes to testing, my choices are macOS and FreeBSD which I do not have experience working with them :)

So I'll close this for now, maybe later open a new PR trying to add tests to those two OSes.

**Comment by felixfontein** [comment]
@aminvakil chances that snap (or any other virtualization technology) will 'just' work on FreeBSD and macOS are pretty much zero :)

**Comment by felixfontein** [comment]
(snap is relying on Linux kernel functionality that's generally not present in non-Linux systems.)

**Comment by aminvakil** [comment]
I clearly have not the slightest idea what `snap` is...

**Comment by felixfontein** [comment]
See https://en.wikipedia.org/wiki/Snap_(package_manager) :) FreeBSD is explicitly mentioned on that page.

It's also pretty similar to flatpak, except that flatpak happily works inside docker containers (at least as long as you don't try to execute programs packaged as flatpak - that probably won't go well either).

**Pull request closed without merging**